src/components/PaymentMethod/PaymentMethod.js

```diff
@@ -31,26 +34,26 @@ const PaymentMethod = ({
       case 'roku':
       case 'googlepay':
       case 'applepay':
-        return (
-          <PaymentCard
-            key={id}
-            details={paymentDetail}
-            showInnerPopup={showInnerPopup}
-          />
-        );
+        return <PaymentCard key={id} details={paymentDetail} />;
       default:
         return <Message>{t('Managed by external service')}</Message>;
     }
   };
   const activeItems = activeOrBoundPaymentDetails.find(item => item.active);
 
+  useEffect(() => {
+    if (paymentDetails?.length === 0) {
+      dispatch(fetchPaymentDetails());
+    }
+  }, []);
+
   return paymentDetailsLoading ? (
     <CardsWrapper numberOfItems={1}>
       <PaymentCard isDataLoaded={false} />
     </CardsWrapper>
   ) : (
     <WrapStyled>
-      {error.length !== 0 ? (
+      {paymentDetailsError.length !== 0 ? (
```

Comment:
just looking into this return function and I think that we could return what we can, as soon as we can, to avoid nesting. So it would be  (starting from line 49): 
```
if(paymentDetailsLoading)
  return (...)
if(paymentDetailsError.length)
  return (...)
//  the component itself

```